### PR TITLE
Add CHP Supplementary Firing

### PR DESCRIPTION
**Pull request by zolanaj**
This PR adds supplementary firing to the REoptLite model.  Key updates: 

- Added relevant decision variables **except** for the binary variable indicating supplementary firing.  Model changes since its original addition rendered it irrelevant as there was a Big-M constraint restricting it but no usage anywhere else in the REopt_Lite_API model. 
- Added output "size_supplementary_firing_kw" to CHP technology
- Updated outputs to include capital cost of supplementary firing
- Set all supplementary firing variables to 0 when inputs preclude its addition in the model
- Updated thermal load constraints and output to include supplementary firing where appropriate (un-commenting and correcting syntax)
- Added a test for supplementary firing

### CODE REVIEW

**Comment by zolanaj** [comment]
@NLaws @Bill-Becker  the standby charges and option to not allow CHP to reduce demand charge are now added here as well.  So, this is now ready for a full review.

File: src/core/chp.jl [hunk 1]
```diff
@@ -112,6 +114,8 @@ Base.@kwdef mutable struct CHP <: AbstractCHP
     supplementary_firing_capital_cost_per_kw::Float64 = 150.0
     supplementary_firing_max_steam_ratio::Float64 = 1.0
     supplementary_firing_efficiency::Float64 = 0.92
+    standby_rate_us_dollars_per_kw_per_month::Float64 = 0.0
+    does_not_reduce_demand_charges::Bool = false
```
Comment (NLaws) [inline src/core/chp.jl:118]: I find the double negative with `does_not_reduce_demand_charges::Bool = false` somewhat confusing. Can we change this to `reduces_demand_charges::Bool = true`? Also, why is this an option? We should document it in the doc string.

File: src/core/reopt.jl [hunk 2]
```diff
@@ -315,6 +325,9 @@ function build_reopt!(m::JuMP.AbstractModel, p::REoptInputs)
 		# Total Fuel Costs, tax deductible for offtaker
         m[:TotalFuelCosts] * (1 - p.s.financial.offtaker_tax_pct) +
 
+		#CHP Standby Charges
+		m[:TotalCHPStandbyCharges] * p.s.financial.offtaker_tax_pct +
```
Comment (NLaws) [inline src/core/reopt.jl:329]: Should this be `m[:TotalCHPStandbyCharges] * (1 - p.s.financial.offtaker_tax_pct)` ?

**Review by NLaws — APPROVED** [review]
Approved conditioned on addressing my two comments.

File: src/core/reopt.jl (hunk 2 above)
Comment (zolanaj) [reply src/core/reopt.jl]: Yes it should, thanks for catching!  Addressed in b794cee968298ae8896cddbc1a9bca76d91d2ae9

File: src/core/chp.jl (hunk 1 above)
Comment (zolanaj) [reply src/core/chp.jl]: I agree with removing the double negative; this has been done in fa909613673b9891a749ecef966c412bf2a0a668

I'll merge this when tests are confirmed, and then get the docs updated in a separate PR.  There are a few things worth documenting on CHP beyond this issue (to which the specific answer is, at least one utility does not allow demand charges to be reduced by running the CHP system, so the default is now "true" but this is something that can be toggled).

**Pull request merged**